pandas/_libs/src/inference.pyx

```diff
@@ -9,6 +9,13 @@ iNaT = util.get_nat()
 
 cdef bint PY2 = sys.version_info[0] == 2
 
+from pandas._libs.tslibs.parsing import (
+    try_parse_dates,
+    try_parse_date_and_time,
+    try_parse_year_month_day,
+    try_parse_datetime_components)
+
```

Comment:
I don't think you actually need to import these here, rather import them directly where they are used